shaker/helpers.py

```diff
@@ -97,3 +98,59 @@ def validate_github_access(response):
                     logging.error("Unknown problem checking credentials: %s" % response_message)
     
     return valid_credentials
+
+def parse_metadata(metadata):
+    """
+    Entry function to handle the metadata parsing workflow and return a metadata 
+    object which is cleaned up
+    
+    Args:
+        metadata (dictionary): Keyed salt formula dependency information
+    
+    Returns:
+        parsed_metadata (dictionary): The original metadata parsed and cleaned up
+    """
+    # Remove duplicates
+    parsed_metadata = resolve_metadata_duplicates(metadata)
+    return parsed_metadata
+    
+def resolve_metadata_duplicates(metadata):
+    """
+    Strip duplicates out of a metadata file. If we have no additional criteria, 
+    simply take the first one. Or can resolve by latest version or preferred organisation
+    if required
+    
+    Args:
+        metadata (dictionary): Keyed salt formula dependency information
+    
+    Returns:
+        resolved_dependencies (dictionary): The original metadata stripped of duplicates
+            If the metadata could not be resolved then we return the original args version
+    """ 
+    # Only start to make alterations if we have a valid metadata format
+    # Otherwise ignore and just return the one we were passed as an argument
+    if metadata and (type(metadata) == type({})) and ("dependencies" in metadata):
```

Comment:
If the metadata is not valid it would be much preferable for the parse function (so `parse_metadata` rather than here) to throw a nice exception rather than returning a string and ending up with an unexpected/hard to diagnose error elsewhere in the code.